src/sentry/static/sentry/app/components/modals/recoveryOptionsModal.jsx

```diff
@@ -58,7 +56,7 @@ class RecoveryOptionsModal extends AsyncComponent {
             {t('You should now set up recovery options to secure your account.')}
           </TextBlock>
 
-          {!skipSms && !smsEnrolled ? (
+          {sms && !sms.isEnrolled && !skipSms ? (
```

Comment:
You repeat this condition in a few places you could hoist it into another variable.

```js
let displaySmsPrompt = sms && !sms.isEnrolled && !skipSms
```